From f71e97d7ae67496c7fd2e0dcdbaec56ab5b0dcc7 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Fri, 28 Jul 2023 18:33:00 +0200
Subject: [PATCH 0420/1141] misc: stm32_omm: fix resources used

Use of devm_platform_ioremap_resource_byname API.
Remove unneeded of_node_put.

Change-Id: I2948118aeb03c64cf8b21e8c93ef824acd3671a3
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
---
 drivers/misc/stm32_omm.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/drivers/misc/stm32_omm.c
+++ b/drivers/misc/stm32_omm.c
@@ -270,7 +270,6 @@ static int stm32_omm_probe(struct platfo
 	struct device_node *child;
 	struct device_node *child_node[OMM_CHILD_NB];
 	struct stm32_omm *omm;
-	struct resource *res;
 	int ret;
 	u8 child_access_granted = 0;
 	u8 i, j;
@@ -282,8 +281,7 @@ static int stm32_omm_probe(struct platfo
 
 	omm->nb_child = 0;
 
-	res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "omm");
-	omm->io_base = devm_ioremap_resource(dev, res);
+	omm->io_base = devm_platform_ioremap_resource_byname(pdev, "omm");
 	if (IS_ERR(omm->io_base))
 		return PTR_ERR(omm->io_base);
 
@@ -298,7 +296,6 @@ static int stm32_omm_probe(struct platfo
 		}
 
 		if (!of_device_is_compatible(child, "st,stm32mp25-omi")) {
-			of_node_put(child);
 			continue;
 		}
 
